### PR TITLE
[SPARK-44246][CONNECT][FOLLOW-UP] Miscellaneous cleanups for Spark Connect Jar/Classfile Isolation

### PR DESCRIPTION
**Pull request by vicennial**
### What changes were proposed in this pull request?

This PR is a follow-up of #41701 and addresses the comments mentioned [here](https://github.com/apache/spark/pull/41701#issuecomment-1608577372). The summary is:

- `pythonIncludes` are directly fetched from the `ArtifactManager` via `SessionHolder` instead of propagating through the spark conf
- `SessionHolder#withContext` renamed to `SessionHolder#withContextClassLoader` to decrease ambiguity.
- General increased test coverage for isolated classloading (New unit test in `ArtifactManagerSuite` and a new suite `ClassLoaderIsolationSuite`.

### Why are the changes needed?

General follow-ups from [here.](https://github.com/apache/spark/pull/41701#issuecomment-1608577372)

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

New test suite and unit tests.

### CODE REVIEW

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 1]
```diff
@@ -1519,7 +1508,7 @@ class SparkConnectPlanner(val sessionHolder: SessionHolder) extends Logging {
       command = fun.getCommand.toByteArray,
       // Empty environment variables
       envVars = Maps.newHashMap(),
-      pythonIncludes = pythonIncludes,
+      pythonIncludes = sessionHolder.artifactManager.getSparkConnectPythonIncludes.asJava,
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:1511]: Gotya. This is exactly the change I planned to made this week :-). Thank you for doing this here.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: core/src/test/scala/org/apache/spark/executor/ClassLoaderIsolationSuite.scala [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.executor
+
+import org.apache.spark.{JobArtifactSet, LocalSparkContext, SparkConf, SparkContext, SparkFunSuite}
+import org.apache.spark.util.Utils
+
+class ClassLoaderIsolationSuite extends SparkFunSuite with LocalSparkContext  {
```
[the remark this answers is not in the document]
Comment (LuciferYang) [reply core/src/test/scala/org/apache/spark/executor/ClassLoaderIsolationSuite.scala:23]: Fix https://github.com/apache/spark/pull/41852

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply core/src/test/scala/org/apache/spark/executor/ClassLoaderIsolationSuite.scala:23]: Thanks @LuciferYang

**Pull request closed without merging**